src/main/java/StringAddCalculator.java

```diff
@@ -0,0 +1,66 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+
+    public static final String CHAR_SLASH = "//";
+    public static final String CHAR_LINE_BREAK = "\\n";
+    public static final String REGEX_DELIMITER = "[,:]";
+
+    public static Integer calculate(String text) {
+        if (isInvalid(text)) {
+            return 0;
+        }
+
+        Integer[] integers = toIntegers(splitText(text));
+        for (Integer integer : integers) {
+            validateInteger(integer);
+        }
+
+        return sum(integers);
+    }
+
+    private static void validateInteger(Integer integer) {
+        if (integer < 0) {
+            throw new RuntimeException();
+        }
+    }
+
+    private static boolean isInvalid(String text) {
+        return text == null || text.isEmpty();
+    }
+
+    private static int sum(Integer[] integers) {
+        int sum = 0;
+        for (Integer integer : integers) {
+            sum += integer;
+        }
+        return sum;
+    }
+
+    private static Integer[] toIntegers(String[] strings) {
+        Integer[] integers = new Integer[strings.length];
+
+        for (int i = 0; i< strings.length; i++) {
+            integers[i] = Integer.parseInt(strings[i]);
+        }
+
+        return integers;
+    }
+
+    private static String[] splitText(String text) {
+        String customDelimiter = extractCustomDelimiter(text);
+        if (customDelimiter != null) {
```

Comment:
```suggestion
        if (delimiter.isCustom()){
```

객체가 있다면, 메시지를 통해 판단을 객체에게 물어볼수있지 않을까요?